es/lessons/basics/documentation.md

```diff
@@ -0,0 +1,309 @@
+
+```elixir
+iex> c("greeter.ex")
+[Greeter]
+
+iex> h Greeter
+
+                Greeter
+
+Provides a function hello/1 to greet a human
+```
+
+### Documentando Funciones
+
+Al igual que Elixir nos da la habilidad para documentar a nivel de módulo, también habilita similares anotaciones para documentar funciones. El anotador `@doc` permite agregar documentación a nivel de función. El anotador `@doc` se situa justo sobre la función que esta comentando.
+
+```elixir
+defmodule Greeter do
+  @moduledoc """
+  ...
+  """
+
+  @doc """
+  Prints a hello message
+
+  ## Parameters
+
+    - name: String that represents the name of the person.
+
+  ## Examples
+
+      iex> Greeter.hello("Sean")
+      "Hello, Sean"
+
+      iex> Greeter.hello("pete")
+      "Hello, pete"
+
+  """
+  @spec hello(String.t) :: String.t
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+```
+
+Si entramos en IEx otra vez y usamos el comando de ayuda (`h`) con la función precedida por el nombre del módulo, veremos lo siguiente:
+
+```elixir
+iex> c("greeter.ex")
+[Greeter]
+
+iex> h Greeter.hello
+
+                def hello(name)
+
+Prints a hello message
+
+Parameters
+
+  • name: String that represents the name of the person.
+
+Examples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+
+iex>
+```
+
+¿Notas como puedes usar marcado dentro de la documentación y la terminal lo renderizará? Además de ser realmente genial y una adición novedosa al vasto ecosistema de Elixir, se vuelve mucho más interesante cuando vemos que ExDoc genera documentación en HTML sobre la marcha.
+
+**Nota:** la anotación `@spec` es usada para analisis estático del código. Para aprender mas acerca de eso, revisa la lección [Especificaciones y tipos](../../advanced/typespec).
+
+## ExDoc
+
+ExDoc es un proyecto oficial de Elixir que puede ser encontrado en [GitHub](https://github.com/elixir-lang/ex_doc). Produce **HTML (HyperText Markup Language) y documentación en linea** para proyectos Elixir. Primero vamos a crear un proyecto Mix para nuestra aplicación:
+
+```bash
+$ mix new greet_everyone
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/greet_everyone.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/greet_everyone_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd greet_everyone
+    mix test
+
+Run "mix help" for more commands.
+
+$ cd greet_everyone
+
+```
+
+Ahora copia y pega el código de la lección del anotador `@doc` en un archivo llamado `lib/greeter.ex` y asegurate de que todo aun esta funcionando desde la linea de comandos. Ahora que estamos trabajando dentro de un proyecto Mix necesitamos empezar IEx de una forma un poco diferente usando el comando `iex -S mix`:
```

Comment:
Asegúrate

Reply:
Línea.
Ahora should be removed from the first sentence and the second one starts with the same word

Reply:
*AS the second one...